pkg/deploy/api/types.go

```diff
@@ -441,6 +441,20 @@ type DeploymentConfigRollbackSpec struct {
 	IncludeStrategy bool
 }
 
+// DeploymentRequest is a request to a deployment config for a new deployment.
+type DeploymentRequest struct {
+	unversioned.TypeMeta
+	// Name of the deployment config for requesting a new deployment.
+	Name string
+	// DryRun if true should not persist the updated deployment config in storage.
+	DryRun bool
+	// Latest will update the deployment config with the latest images found in the cluster.
+	Latest bool
```

Comment:
> If a controller is paused and I call this, what happens?

Good question. We may want to "force" a deployment even when paused but I am not sure about it. We already disallow creating new controllers for paused configs in the deploymentconfig controller. If somebody wants real manual deployments then they should not use CC triggers and any IC should be non-automatic. WDYT?